### PR TITLE
Add secure and samesite options

### PR DESCRIPTION
**Pull request by Demetri0**
Provide ability to set `secure` and `samesite` options for cookies

#28

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ export const setCookie = (name, value, options) => {
   const optionsWithDefaults = {
     days: 7,
     path: '/',
+    samesite: 'none'
+    secure: false,
```
Comment (tylerwolff) [inline index.js:12]: sorry havent looked at this repo in too long, but I'm making updates now. However, I think for this we might want to default to the browser default `samesite: 'lax'` https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#lax

Comment (tylerwolff) [reply index.js:12]: I'll just update it, just wanted to let you know

Comment (Demetri0) [reply index.js:12]: OK, Good to to see you back :) Thank you. I missed that `lax` started to be default

**Comment by elibroftw** [comment]
This needs to be added asap because a) security and b) Firefox is complaining about SameSite: None

File: index.d.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ declare module 'react-use-cookie' {
   interface cookieOptions {
     days?: number;
     path?: string;
+    samesite?: 'none' | 'lax' | 'strict';
```
Comment (elibroftw) [inline index.d.ts:10]: Users should be recommended to use useful values:
`'lax' | 'strict' | 'none'`
Also add a link to https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite

File: index.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ export const setCookie = (name, value, options) => {
   const optionsWithDefaults = {
     days: 7,
     path: '/',
+    samesite: 'none'
```
Comment (elibroftw) [inline index.js:11]: Default should be `'lax'`

File: README.md [hunk 4]
```diff
@@ -73,9 +73,13 @@ above:
 ```tsx
 {
   // The number of days the cookie is stored (defaults to 7)
-  days: number;
+  days?: number;
   // The path of the cookie (defaults to '/')
-  path: string;
+  path?: string;
+  // SameSite prevents the browser from sending this cookie along with cross-site requests. Possible values are lax, strict or none
+  samesite?: 'none' | 'lax' | 'strict';
```
Comment (elibroftw) [inline README.md:80]: See other comment about order of possible values.

**Review by elibroftw — CHANGES_REQUESTED** [review]

**Comment by tylerwolff** [comment]
Hey @elibroftw and @Demetri0, sorry for being so late on this but I just pushed v1.4.0 which adds the ability to set other cookie options like `SameSite`. Try it out and let me know if it's working for you.

**Comment by elibroftw** [comment]
Is there a reason SameSite and Secure are capitlized? I'll use the module, but I think we should at least think about the convention we want to set. I had gone with `js-cookies` when I made my review, and its API is like this:
```js
Cookies.set('colorScheme', value, { expires: 365000, sameSite: 'lax', path: '/' });
```
sameSite is not SameSite here.

**Comment by tylerwolff** [comment]
@elibroftw good point, I was just going off how they do it in the MDN docs https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie, but we can do it anyway since browsers don't really care.

**Comment by Demetri0** [comment]
So, because of this feature was implemented - I'm closing MR.

**Pull request closed without merging**